src/tests/ftest/util/fault_config_utils.py

```diff
@@ -211,34 +214,70 @@ class FaultInjectionFailed(Exception):
     """Raise if FI failed."""
 
 
-def write_fault_file(path, fault_list=None, on_the_fly_fault=None):
-    """ Write out a fault injection config file.
+class FaultInjection(Test):
+    """Fault Injection
 
-        path             --where to write the file
-        fault_list       --a list of strings identifying which predefined
-                            faults to write out
-        on_the_fly_fault --a fault dictionary that isn't predefined
-
-        Returns the name of the file.
+    :avocado: recursive
     """
-    if fault_list is None and on_the_fly_fault is None:
-        raise FaultInjectionFailed("bad parameters")
-
-    if not os.path.exists(path):
-        os.makedirs(path)
-    fi_config = path + '/fi.yaml'
-
-    with io.open(fi_config, 'w', encoding='utf8') as outfile:
-        yaml.dump({'seed': '123'}, outfile, default_flow_style=False,
-                  allow_unicode=True)
-
-        fault_config = []
-        if fault_list is not None:
-            for fault in fault_list:
-                fault_config.append(FAULTS[fault])
-        if on_the_fly_fault is not None:
-            fault_config.append(on_the_fly_fault)
-        yaml.dump({'fault_config': fault_config}, outfile,
-                  default_flow_style=False, allow_unicode=True)
-
-    return fi_config
+    def __init__(self, test, fault_file):
+        super().__init__(test, fault_file)
+        self._test = test
+        self._hosts = None
+        self._fault_file = fault_file
+
+    def write_fault_file(path, fault_list=None, on_the_fly_fault=None):
+        """ Write out a fault injection config file.
+
+            path             --where to write the file
+            fault_list       --a list of strings identifying which predefined
+                                faults to write out
+            on_the_fly_fault --a fault dictionary that isn't predefined
+
+            Returns the name of the file.
+        """
+        if fault_list is None and on_the_fly_fault is None:
+            raise FaultInjectionFailed("bad parameters")
+
+        if not os.path.exists(path):
+            os.makedirs(path)
+        fi_config = path + '/fi.yaml'
```

Comment:
This requires moving `self.test_dir = os.getenv("DAOS_TEST_LOG_DIR", "/tmp")` from `TestWithoutServers.__init__()` to `Test.__init__()`:
```suggestion
        fi_config = os.path.join(self.test_dir, "fi.yaml")
```

Reply:
Done.